velox/exec/Exchange.h

```diff
@@ -209,20 +214,26 @@ class ExchangeSource : public std::enable_shared_from_this<ExchangeSource> {
   using Factory = std::function<std::shared_ptr<ExchangeSource>(
       const std::string& taskId,
       int destination,
-      std::shared_ptr<ExchangeQueue> queue)>;
+      std::shared_ptr<ExchangeQueue> queue,
+      memory::MemoryPool* pool)>;
```

Comment:
@tanjialiang This is a nice change as we discussed earlier. Let's move it into a separate PR to make this PR smaller and easier to review.

Reply:
Hi @mbasmanova There is a PR already that has this change. After landing that PR and rebasing this one. The changes will be gone from this PR. I'm manually pasting the chagnes just to get this PR out.

Reply:
Got it. Found #2447.